src/generate/register.rs

```diff
@@ -50,36 +51,62 @@ pub fn render(
 
     let can_read = [Access::ReadOnly, Access::ReadWriteOnce, Access::ReadWrite].contains(&access);
     let can_write = access != Access::ReadOnly;
+    let can_reset = res_val.is_some();
 
     if can_read {
         let desc = format!("Reader of register {}", register.name);
         mod_items.extend(quote! {
             #[doc = #desc]
-            pub type R = crate::R<#rty, super::#name_pc>;
+            pub struct R(crate::R<#name_uc_spec>);
+
+            impl core::ops::Deref for R {
+                type Target = crate::R<#name_uc_spec>;
+
+                fn deref(&self) -> &Self::Target {
```

Comment:
Add `#[inline(always)]` for each `Deref` and `DerefMut` first and retest.
Do you use `cargo size` from `cargo-binutils` or something else?

Reply:
That did it!  Now it's exactly the same size.

These numbers are from my system `avr-size`.  I just tried installing `cargo-binutils` and I'm getting 'Failed to parse crate metadata'.